dev/devicelab/lib/framework/ab.dart

```diff
@@ -2,23 +2,102 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert';
+import 'dart:io';
 import 'dart:math' as math;
+
 import 'package:meta/meta.dart';
+import 'package:args/args.dart';
+
+const String kBenchmarkTypeKeyName = 'benchmark results type';
+const String kBenchmarkVersionKeyName = 'version';
+const String kLocalEngineKeyName = 'local engine name';
+const String kTaskNameKeyName = 'task name';
+const String kRunStartKeyName = 'run start date';
+const String kRunEndKeyName = 'run end date';
+const String kAResultsKeyName = 'default engine results';
+const String kBResultsKeyName = 'local engine results';
+
+const String kBenchmarkResultsType = 'A/B summaries';
+const String kBenchmarkABVersion = '1.0';
+
+void _usage(String error) {
+  stderr.writeln(error);
+  stderr.writeln('Usage:\n');
+  stderr.writeln(_argParser.usage);
+  exitCode = 1;
+}
+
+Future<void> main(List<String> rawArgs) async {
```

Comment:
Would it be possible to move the `main` under `bin/`? The Dart VM doesn't like entrypoints under `lib/`. It's also a pub convention (e.g. you can `pub global activate` an executable from `bin/`).

Reply:
Would also be super-useful to have some docs in the README.md about how to use this.

Reply:
I'll do that. Also, I print the name of the output file at the end of a run - should I mention the "processing" program there too?

Reply:
tool code moved to `bin/summarize.dart`
`README.md` updated